src/main/java/org/wso2/carbon/inbound/salesforce/poll/LoginHelper.java

```diff
@@ -0,0 +1,185 @@
+ */
+package org.wso2.carbon.inbound.salesforce.poll;
+
+import java.io.ByteArrayInputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.ConnectException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.ByteBufferContentProvider;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ * A helper to obtain the Authentication bearer token via login
+ *
+ * @author hal.hildebrand
+ * @since 202
+ */
+public class LoginHelper {
+
+    private static class LoginResponseParser extends DefaultHandler {
+
+        private String buffer;
+        private String faultstring;
+        private boolean reading = false;
+        private String serverUrl;
+        private String sessionId;
+
+        @Override
+        public void characters(char[] ch, int start, int length) {
+
+            if (reading)
+                buffer = new String(ch, start, length);
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) {
+
+            reading = false;
+            switch (localName) {
+                case "sessionId":
+                    sessionId = buffer;
+                    break;
+                case "serverUrl":
+                    serverUrl = buffer;
+                    break;
+                case "faultstring":
+                    faultstring = buffer;
+                    break;
+                default:
+            }
+            buffer = null;
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attributes) {
+
+            switch (localName) {
+                case "sessionId":
+                    reading = true;
+                    break;
+                case "serverUrl":
+                    reading = true;
+                    break;
+                case "faultstring":
+                    reading = true;
+                    break;
+                default:
+            }
+        }
+    }
+
+    public static final String COMETD_REPLAY = "/cometd/";
+    public static final String COMETD_REPLAY_OLD = "/cometd/replay/";
+    static final String LOGIN_ENDPOINT = "https://login.salesforce.com";
+    private static final String ENV_END = "</soapenv:Body></soapenv:Envelope>";
+    private static final String ENV_START =
+            "<soapenv:Envelope xmlns:soapenv='http://schemas.xmlsoap.org/soap/envelope/' "
+                    + "xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' "
+                    + "xmlns:urn='urn:partner.soap.sforce.com'><soapenv:Body>";
+    // The enterprise SOAP API endpoint used for the login call

```

Comment:
Leave an empty line before this single line comment.

Reply:
line added